src/compiler/builder.ts

```diff
@@ -193,6 +204,19 @@ namespace ts {
             return;
         }
 
+        // Clean lib file diagnostics if its all files excluding default files to emit
+        if (state.allFilesExcludingDefaultLibraryFile === state.affectedFiles && !state.cleanedDiagnosticsOfLibFiles) {
+            state.cleanedDiagnosticsOfLibFiles = true;
+            const options = state.program.getCompilerOptions();
+            if (forEach(state.program.getSourceFiles(), f =>
+                !contains(state.allFilesExcludingDefaultLibraryFile, f) &&

```

Comment:
This seems like it could be slow. For *each* affected file, for *each* file in getSourceFiles(), for *each* file in `allFilesExcludingDefaultLibraryFile` we do work.
We're testing that `f` *isn't* contained in the list of files that *aren't* the default library file -- isn't that equivalent to testing whether `f` *is* the default library file? But then instead of looping over all files and testing whether it is the default library file, we could just do this for the default library file..